### PR TITLE
Fix hive table can not join reorder

### PR DESCRIPTION
**Pull request by Youngwb**
For Dp and greedy join reorder，we check the column statistics is unknown firstly. For olap scan and hive scan, there are different column statistcs cache.  we should use hive table columns statistics cache to get it own column statistics

### CODE REVIEW

**Comment by imay** [comment]
@caneGuy Need IcebergTable to be checked?

File: fe/fe-core/src/main/java/com/starrocks/external/hive/HiveColumnStats.java [hunk 1]
```diff
@@ -25,6 +25,11 @@
     public HiveColumnStats() {
     }
 
+    public boolean isDefaultValue() {
+        return avgSize == -0.1f && numNulls == -1L && numDistinctValues == -1L && minValue == Double.NEGATIVE_INFINITY
```
Comment (dirtysalt) [inline fe/fe-core/src/main/java/com/starrocks/external/hive/HiveColumnStats.java:29]: Could there be a field 'bool unknwon' to represent the stats is unknown?

Comment (Youngwb) [reply fe/fe-core/src/main/java/com/starrocks/external/hive/HiveColumnStats.java:29]: Yes, I think we  should add an Unknown type later，and set unknown if we can not get column statsitics from meta

Comment (Youngwb) [reply fe/fe-core/src/main/java/com/starrocks/external/hive/HiveColumnStats.java]: Done

**Review by dirtysalt — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/external/hive/HiveColumnStats.java [hunk 2]
```diff
@@ -161,6 +171,9 @@ public boolean init(String hiveType, ColumnStatisticsData statsData) {
                 LOG.warn("unexpected column type {}", hiveType);
                 break;
         }
+        if (isCompatible) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/external/hive/HiveColumnStats.java:174]: what's the meaning of `isCompatible` ?

Comment (Youngwb) [reply fe/fe-core/src/main/java/com/starrocks/external/hive/HiveColumnStats.java:174]: @gengjun-git

Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/external/hive/HiveColumnStats.java:174]: isCompatible means the stats data is valid

Comment (kangkaisen) [reply fe/fe-core/src/main/java/com/starrocks/external/hive/HiveColumnStats.java:174]: So why use `Compatible` to mean `valid`?  I think which is incomprehensible.

**Review by dirtysalt — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by caneGuy** [comment]
> @caneGuy Need IcebergTable to be checked?

yes, this is related with https://github.com/StarRocks/starrocks/pull/2805

**Pull request merged**